src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,332 @@
+import click
+import contextlib
+import errno
+import json
+import logging
+import os
+import pathlib
+import subprocess
+import sys
+import tempfile
+import traceback
+from typing import List, Tuple, Optional
+import urllib.request
+
+if sys.platform.startswith("win"):
+   import winreg
+
+MODULE_DIR = pathlib.Path(__file__).parent
+METADATA_FILE = MODULE_DIR / "metadata.json"
+
+def _parse_version(version: str) -> Tuple[int, ...]:
+    """
+    Split the version string into a tuple of integers representing major, minor, and update parts.
+
+    Args:
+        version (str): The version string in the format "major.minor.update".
+
+    Returns:
+        tuple: A tuple of integers representing the major, minor, and update parts of the version.
+
+    >>> _parse_version("23.8.0")
+    (23, 8, 0)
+    >>> _parse_version("24.0.0")
+    (24, 0, 0)
+    >>> _parse_version("invalid_version")
+    Traceback (most recent call last):
+    ...
+    click.exceptions.ClickException: Invalid version format found
+    """
+    try:
+        return tuple(int(part) for part in version.split("."))
+    except ValueError  as e:
+        logging.info("Error: %s", e)
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'Invalid version format found') from e
+
+def _is_latest_version_greater(latest_version : str, installed_version: str) -> int:
+    """
+    Compare two versions which are in the format: major.minor.update
+
+    Args:
+        installed_version (str): The installed version to compare.
+        latest_version (str): The version to compare against.
+
+    Returns:
+        int: False if versions are equal and installed_version is greater than latest_version
+        - True if installed_version is less than latest_version,
+             .
+
+    >>> _is_latest_version_greater("23.8.0", "23.8.0")
+    Latest Version already installed
+    False
+    >>> _is_latest_version_greater("23.8.1", "24.0.0")
+    Latest Version already installed
+    False
+    >>> _is_latest_version_greater("23.8.1", "23.8.0")
+    True
+    >>> _is_latest_version_greater("24.0.0", "23.8.5")
+    True
+    >>> _is_latest_version_greater("invalid_version", "1.2.0")
+    Traceback (most recent call last):
+    ...
+    click.exceptions.ClickException: Invalid version format found
+    """
+    try:
+        logging.debug("Entering _compare_version")
+        latest_parts: Tuple[int, ...] = _parse_version(latest_version)
+        installed_parts: Tuple[int, ...] = _parse_version(installed_version)
+
+        if latest_parts > installed_parts:
+            return True
+        else:
+            print(f"Latest Version already installed")
```

Comment:
Maybe give a bit more information here...

```suggestion
            print(f"Installed NI-DAQmx version ({installed_version}) is equivalent or newer than the known version to install, {latest_parts}.")
```